magento/magento2
# Fix negative credit memo #19899

### Description

This fixes negative credit memo problem described in #19899. The problem looks as introduced in \Magento\Sales\Model\Order\Creditmemo\Total\Discount which calculates base_shipping_amount based on tax related variables if base_shipping_amount is zero:

    private function getBaseShippingAmount(\Magento\Sales\Model\Order\Creditmemo $creditmemo): float
    {
        $baseShippingAmount = (float)$creditmemo->getBaseShippingAmount();

        if (!$baseShippingAmount) {
            $baseShippingInclTax = (float)$creditmemo->getBaseShippingInclTax();
            $baseShippingTaxAmount = (float)$creditmemo->getBaseShippingTaxAmount();
            $baseShippingAmount = $baseShippingInclTax - $baseShippingTaxAmount;
        }

        return $baseShippingAmount;
    }

I fixed this by setting base_shipping_incl_tax to 0 in \Magento\Sales\Model\Order\CreditmemoFactory::initData too:

    $creditmemo->setBaseShippingInclTax((double)$data['shipping_amount']);

### Fixed Issues (if relevant)

1. #19899

### Manual testing scenarios

1. Create Cart Price Rule with 100% discount, applied to shipping, available for everyone
2. Create simple product with positive stock and price = $100
3. Place order in backend for the created product with qty =1 and flat rate shipping ($5 by default).
4. Invoice order (but don't create shipment)
5. Navigate to Credit Memo form
6. Change Refund Shipping from 5 to 0
7. Click "Refund Offline"

Expected behavior is to have $0 credit memo created. Instead it creates credit memo with negative, -$5.0 total.

### Contribution checklist
